hieroplatform/src/main/java/org/hiero/sketch/table/HashSubSchema.java

```diff
@@ -49,4 +52,9 @@ public boolean isColumnPresent(final String name) {
     }
 
     public int getColumnCount() { return this.colNames.size(); }
+
+    @Override
+    public Iterator<String> iterator() {
```

Comment:
We have to be careful with this, since this order is non-deterministic.
The schema iterators and column orders in tables are all deterministic. 
This is actually important for users.